src/api/ledger/parse/pathfind.js

```diff
@@ -8,17 +8,25 @@ function parsePaths(paths) {
     _.omit(step, ['type', 'type_hex'])));
 }
 
+function ensureIOUCounterparty(amount, sourceAmount) {
+  const defaults = sourceAmount.counterparty ?
+    {counterparty: sourceAmount.counterparty} : {};
+  return amount.currency === 'XRP' || amount.counterparty ? amount :
+    _.assign(defaults, amount);
+}
+
 function parsePathfind(sourceAddress: string,
     destinationAmount: Object, pathfindResult: Object): Object {
   return pathfindResult.alternatives.map(function(alternative) {
+    const sourceAmount = parseAmount(alternative.source_amount);
     return {
       source: {
         address: sourceAddress,
-        amount: parseAmount(alternative.source_amount)
+        amount: sourceAmount
       },
       destination: {
         address: pathfindResult.destination_account,
-        amount: destinationAmount
+        amount: ensureIOUCounterparty(destinationAmount, sourceAmount)
```

Comment:
> counterparty wouldn't even be set?

Which wouldn't serialise. 

One sense of `never be exposed` means removing functionality. If you're ok with that, then you should just demand a counterparty for IOU destination amounts. However, it's the only way you can send $100 across multiple trust lines, no?  A Payment's paths is specified not only by Paths but by Amount and SendMax I believe. (Also, what about old transactions loaded into the new format? See: https://github.com/ripple/ripple-lib/blob/develop/src/api/ledger/parse/payment.js#L38-L44)